src/ToDo/Business/Services/AuthenticationService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Text.Json;
+using Microsoft.Identity.Client;
+using ToDo.Business.Entities;
+using Uno.UI.MSAL;
+
+namespace ToDo.Business.Services
+{
+	public class AuthenticationService: IAuthenticationService
+	{
+		//public UserContext _userContext;
+		private readonly IPublicClientApplication _pca;
+		public AuthenticationService()
+		{
+			_pca = PublicClientApplicationBuilder
+					.Create(OAuthSettings.ApplicationId)
+					.WithRedirectUri(OAuthSettings.RedirectUri)
+					.WithUnoHelpers()
+#if __IOS__
+                .WithIosKeychainSecurityGroup("9TB54R6A6V.com.horus.unoplatform.todoapp")
+#endif
+					.Build();
+		}
+
+		public async Task<string> LoginAsync()
+		{
+			try
+			{
+				var authResult = await _pca.AcquireTokenInteractive(OAuthSettings.Scopes.Split(' '))
+					.WithUnoHelpers()
+					.ExecuteAsync();
+				//TODO:We need to store UserContext in order to use it in the HomeViewModel
+				var userContext = CreateContextFromAuthResult(authResult);
+				return authResult.AccessToken;
+			}
+			catch (MsalClientException ex)
+			{
+				//This is thrown when the user closes the webview before he can authenticate
+				throw new MsalClientException(ex.ErrorCode, ex.Message);
+			}
+			catch (Exception ex)
+			{
+				throw new Exception(ex.Message);
+			}
+			
+		}
+
+		private UserContext CreateContextFromAuthResult(AuthenticationResult authResult)
+		{
+			return ParseToken(authResult.IdToken);
+		}
+
+		UserContext ParseToken(string idToken)
+		{
+			var data = idToken.Split('.')[1];
+			idToken = Base64UrlDecode(data);
```

Comment:
Can you check to see if we can use JwtSecurityTokenHandler in order to decord the idToken, instead of manually decoding it